lib/init/npm-utils.js

```diff
@@ -119,12 +119,12 @@ function check(packages, opt) {
         throw error;
     }
 
-    if (opt.devDependencies && typeof fileJson.devDependencies === "object") {
-        deps = deps.concat(Object.keys(fileJson.devDependencies));
-    }
-    if (opt.dependencies && typeof fileJson.dependencies === "object") {
-        deps = deps.concat(Object.keys(fileJson.dependencies));
-    }
+    ["dependencies", "devDependencies"].forEach(dependency => {
```

Comment:
I wonder if it would be it would be clearer to rename `dependency` to `key` here. `dependency` signals to me that it's iterating over a list of dependencies rather than a list of JSON key names.

```suggestion
    ["dependencies", "devDependencies"].forEach(key => {
```